project-base/storefront/urql/operationNameExchange.ts

```diff
@@ -0,0 +1,45 @@
+import { Kind, OperationDefinitionNode } from 'graphql';
+import { getStringWithoutTrailingSlash } from 'helpers/parsing/stringWIthoutSlash';
+import { Exchange } from 'urql';
+import { pipe, tap } from 'wonka';
+
+export const operationNameExchange: Exchange =
+    ({ forward }) =>
+    (ops$) => {
+        return pipe(
+            ops$,
+            tap((operation) => {
+                const operationName =
+                    (
+                        operation.query.definitions.find(
+                            (definition) => definition.kind === Kind.OPERATION_DEFINITION,
+                        ) as OperationDefinitionNode | undefined
+                    )?.name?.value ?? 'UnknownOperation';
+
+                const urlWithoutTrailingSlash = getStringWithoutTrailingSlash(operation.context.url);
+                const urlWithOperationName = urlWithoutTrailingSlash + `/${operationName}`;
+
+                let existingHeaders: HeadersInit | undefined = {};
+                if (operation.context.fetchOptions) {
+                    if (typeof operation.context.fetchOptions === 'function') {
+                        existingHeaders = operation.context.fetchOptions().headers;
+                    } else {
+                        existingHeaders = operation.context.fetchOptions.headers;
+                    }
+                }
```

Comment:
Since this variable will be always defined what about keep the logic for it within closure? I think code would be little cleaner.

```suggestion
                const existingHeaders = (() => {
                  if (!operation.context.fetchOptions) {
                     return undefined
                  }
                  
                  if (typeof operation.context.fetchOptions === 'function') {
                          return operation.context.fetchOptions().headers;
                  }
                      
                  return operation.context.fetchOptions.headers;
                })();
```

Reply:
Sure